tools/emptyDropsWrapper/emptyDropsWrapper.R

```diff
@@ -0,0 +1,206 @@
+                default=FALSE, ## as per emptyDrops package
+                action='store_true',
+                help='Empty drops parameter test.ambient'),
+    make_option(c('--emptydrops-ignore'),
+                default=NULL, ## as per emptyDrops package
+                help='emptyDrops ignore parameter',
+                dest='ed_ignore'),
+    make_option(c('--emptydrops-alpha'),
+                default=NULL, ## as per emptyDrops package
+                help='emptyDrops alpha parameter',
+                dest='ed_alpha')
+    ## TODO: Expose parallel functionality
+)
+
+## Parse the arguments
+opt_parser <- OptionParser(option_list=option_list);
+opt <- parse_args(opt_parser);
+
+## Check the parsed arguments
+if(is.null(opt$input_rds))  errorExit("Input RDS is not specified\n");
+if(is.null(opt$output_csv)) errorExit("Output CSV is not specified\n");
+if(!file.exists(opt$input_rds)) errorExit("Input RDS doesn't exist!\n");
+if(file.exists(opt$output_csv)) errorExit("Output CSV file exists!\n");
+
+## Load the required libraries here
+## NOTE: We do this after parsing arguments so that --help returns immediatedly
+pkgs <- c('DropletUtils','Matrix')
+missing_pkgs <- pkgs[!pkgs %in% installed.packages()]
+if(length(missing_pkgs) > 0) {
+    cat('Error the following package(s) are missing:', paste0(missing_pkgs, collapse=', '),'\n')
+    quit(save="no",status=1)
+}
+
+## Load the libs, suppressing warnings and other messages
+suppressWarnings(suppressPackageStartupMessages(library('DropletUtils',quietly=TRUE)))
+suppressWarnings(suppressPackageStartupMessages(library('Matrix',quietly=TRUE)))
+
+## Set script parameters, this need to be done after package loading
+## as SerialParam() is not otherwise availbale
+inputRDS <- opt$input_rds
+output_csv <- opt$output_csv
+FDRcutoff <- opt$fdr_cutoff
+verbose <- opt$verbose
+ed_param_lower <- opt$verbose
+ed_param_niters <- opt$ed_niters
+ed_param_test.ambient <- opt$ed_test_ambient
+ed_param_ignore <- opt$ed_ignore
+ed_param_alpha <- opt$ed_alpha
+ed_param_BPPARAM=SerialParam() ## TODO: Allow serial or parallele, with ncore specification
+
+## Read the input file
+inputMatrix <- readRDS(inputRDS)
+
+## Check the input matrix
+## Check class
+if(!class(inputMatrix) %in% c( 'dgCMatrix','dgRMatrix' )) {
+    cat(paste0('Error: input matrix is not of class dgCMatrix or dgRMatrix. It is of class:',class(inputMatrix)),file=stderr())
+    quit(save="no",status=1)
+}
+## Check dimensions
+if(any(dim(inputMatrix) == 0)) {
+    cat('Error: one or more dimensions of the input matrix are empty',file=stderr())
+    quit(save="no",status=1)
+}
+## Check row and column names
+if(is.null(rownames(inputMatrix))) {
+    cat('Warning: rownames of the input matrix are empty',file=stderr())
+    quit(save="no",status=1)
+}
+if(is.null(colnames(inputMatrix))) {
+    cat('Warning: colnames of the input matrix are empty',file=stderr())
+    quit(save="no",status=1)
+}
+
+## Convert to CsparseMatrix if required
```

Comment:
Can you please remove this block of commented out code if it is no longer needed?

Reply:
I left it on purpose because its difficult to find it again and it might be useful during optimization of this script. Is that acceptable or should I remove it anyway?